core-cube/src/main/java/org/apache/kylin/cube/CubeInstance.java

```diff
@@ -77,6 +82,8 @@
     public static final int COST_WEIGHT_DIMENSION = 10;
     public static final int COST_WEIGHT_INNER_JOIN = 100;
 
+    private volatile IRealization[] realizations = null;
```

Comment:
This is not needed. Please narrow down the scope of a variable as much as possible.